networkx/algorithms/operators/binary.py

```diff
@@ -329,3 +329,55 @@ def compose(G, H):
     else:
         R.add_edges_from(H.edges(data=True))
     return R
+
+
+def full_join(G, H, rename=(None, None), name=None):
+    """Returns the full join of graphs G and H.
+
+    Full join is the union of G and H in which all edges between
+    G and H are added.
+    The node sets of G and H must be disjoint,
+    otherwise an exception is raised.
+
+    Parameters
+    ----------
+    G, H : graph
+       A NetworkX graph
+
+    rename : bool , default=(None, None)
+       Node names of G and H can be changed by specifying the tuple
+       rename=('G-','H-') (for example).  Node "u" in G is then renamed
+       "G-u" and "v" in H is renamed "H-v".
+
+    name : string
+       Specify the name for the union graph
+
+    Returns
+    -------
+    U : The full join graph with the same type as G.
+
+    Notes
+    -----
+    It is recommended that G and H be either both directed or both undirected.
+
+    The full join operation of graphs G and H is the same as getting
+    their complement, performing a disjoint union, and finally getting
+    the complement of the resulting graph.
+
+    Graph, edge, and node attributes are propagated from G and H
+    to the union graph.  If a graph attribute is present in both
+    G and H the value from H is used.
+
+    See Also
+    --------
+    union
+    """
+    R = union(G, H, rename, name)
```

Comment:
This full_join operation looks like it adds selfloops for nodes of H (but not selfloops for nodes of G). 
Also, I thought the ```union``` operation would add edges from H into R. Do we need to add the edges from H below?  You can use ```list(R)``` instead of ```list(R.nodes())```. Finally, does this assume undirected graphs? Do we need to handle directed graphs (or multigraphs) specially? You can use ```not_implemented_for``` decorators to restrict usage to undirected and to rule out multiedges.

Reply:
Fixed the possibility of adding selfloops and removed the redundant addition of edges from H into R.

Reply:
But it adds edges between every two nodes in H...  And why do you need to compute nodes?
Would this work?   

    R = nx.union(G, H, rename, name)
    for i in G:
        for j in H:
            R.add_edge(i, j)
    if R.is_directed():
        for i in H:
            for j in G:
                R.add_edge(j, i)

Actually, don't we have to worry about "rename" and adding a prefix to each node name?
And maybe we should just not include the "name" argument to the union command. People don't use that very often and they can easily do it by hand if they want it. We're moving away from keeping "name" as a special attribute.

Reply:
Yes. This works.